packages/key-management/src/util/ownSignatureKeyPaths.ts

```diff
@@ -68,18 +94,25 @@ export const ownSignatureKeyPaths = async (
   txBody: Cardano.TxBody,
   knownAddresses: GroupedAddress[],
   inputResolver: Cardano.util.InputResolver
-): Promise<AccountKeyDerivationPath[]> => {
+): Promise<SignatureKeyPathsResult> => {
+  let partialSign = false;
+
   const paymentKeyPaths = uniq(
     (
       await Promise.all(
         txBody.inputs.map(async (input) => {
           const ownAddress = await inputResolver.resolveInputAddress(input);
           if (!ownAddress) return null;
-          return knownAddresses.find(({ address }) => address === ownAddress);
+
+          const foundAddress = knownAddresses.find(({ address }) => address === ownAddress);
+
+          if (foundAddress === undefined) partialSign = true;
+
+          return foundAddress;
         })
       )
     ).filter(isNotNil)
   ).map(({ type, index }) => ({ index, role: Number(type) }));
 
-  return [...paymentKeyPaths, ...getStakingKeyPaths(knownAddresses, txBody)];
+  return { keyPaths: [...paymentKeyPaths, ...getStakingKeyPaths(knownAddresses, txBody)], partialSign };
```

Comment:
`partialSign` should also be `true` if there are certificates that require signature for a foreign stake key